WordPress/Classes/Services/PostRepository.swift

```diff
@@ -225,6 +225,7 @@ final class PostRepository {
     @MainActor
     private func _create(_ post: AbstractPost, changes: RemotePostUpdateParameters?) async throws -> RemotePost {
         let service = try getRemoteService(for: post.blog)
+        post.foreignID = UUID().uuidString
```

Comment:
This `post` is a revision and this is why `[blog lookupPostWithForeignID:remotePost.foreignID inContext:context];` fails. If you remove `original == NULL` from the predicate, it works.

P.S. yeah it's a bit unexpected, but the new posts that were never uploaded still have an (empty) original post and a one or more local revisions.

Reply:
Fixed [74baced](https://github.com/wordpress-mobile/WordPress-iOS/pull/23173/commits/74baced31828a3450aabd41f1292928bdf9ee38d)